tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,294 @@
+    }
+
+    # Create project
+    assert PROJECT.new_project(TEST_PROJECT)
+
+    # Create storage class
+    create_storage_class(**sc_parms)
+
+    # Create 100 PVCs
+    create_multiple_pvc(self.pvc_base_name, self.number_of_pvc)
+
+    # Verify PVCs are Bound
+    for count in range(1, self.number_of_pvc + 1):
+        pvc_name = f'{self.pvc_base_name}{count}'
+        pv = verify_pvc_and_fetch_pv_name(pvc_name)
+        assert pv, f'PVC {pvc_name} does not exists'
+        self.initial_pvs.append(pv)
+
+
+def teardown(self):
+    """
+    Delete PVCs
+    Delete storage class
+    Delete temporary yaml files
+    Delete project
+    """
+    # Delete newly created PVCs
+    command = (
+        f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
+        f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done'
+    )
+    proc = run_async(command)
+    assert proc, (
+        f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
+    )
+
+    # Verify command to delete PVCs
+    ret, _, _ = proc.async_communicate()
+    assert not ret, "Deletion of newly created PVCs failed"
+
+    # Delete storage class
+    assert SC.delete(yaml_file=TEMP_YAML_FILE_SC)
+
+    # Switch to default project
+    run_cmd(f'oc project {defaults.ROOK_CLUSTER_NAMESPACE}')
+
+    # Delete project created for the testcase
+    PROJECT.delete(resource_name=TEST_PROJECT)
+
+    delete_file(TEMP_YAML_FILE_SC)
+
+
+def create_storage_class(**kwargs):
+    """
+    Create storage class
+
+    Args:
+        **kwargs: key, value pairs for yaml file substitution
+    """
+    sc_yaml = os.path.join(TEMPLATE_DIR, "storageclass.yaml")
+    sc_name = kwargs['storageclass_name']
+
+    sc_yaml_file = templating.generate_yaml_from_jinja2_template_with_data(
+        sc_yaml, **kwargs
+    )
+    with open(TEMP_YAML_FILE_SC, 'w') as yaml_file:
+        yaml.dump(sc_yaml_file, yaml_file, default_flow_style=False)
+    log.info(f'Creating Storage Class {sc_name}')
+    assert SC.create(yaml_file=TEMP_YAML_FILE_SC)
+    log.info(f'Created Storage Class {sc_name}')
+
+
+def create_multiple_pvc(pvc_base_name, number_of_pvc):
+    """
+    Create PVCs
+
+    Args:
+        pvc_base_name (str): Prefix of PVC name
+        number_of_pvc (int): Number of PVCs to be created
+    """
+    for count in range(1, number_of_pvc + 1):
+        pvc_name = f'{pvc_base_name}{count}'
+        log.info(f'Creating Persistent Volume Claim {pvc_name}')
+        PVOLC.name = pvc_name
+        assert PVOLC.create_pvc(storageclass='test-sc')
+        log.info(f'Created Persistent Volume Claim {pvc_name}')
+
+
+def run_async(command):
+    """
+    Run command locally and return without waiting for completion
+
+    Args:
+        command (str): The command to run on the system.
+
+    Returns:
+        An open descriptor to be used by the calling function.
+        None on error.
+    """
+    log.info(f"Executing command: {command}")
+    p = subprocess.Popen(
```

Comment:
@clacroix12 I think we can move your run_cmd in deployment to utils so that folks can use that?

Reply:
It already is. https://github.com/red-hat-storage/ocs-ci/blob/master/utility/utils.py#L626

Reply:
Thanks!

Reply:
@jilju you can use the above method here.

Reply:
@vasukulkarni This method  is to run a command locally and return an open descriptor (without waiting for completion of the command) which can be used to check the status of the command at a later point of time.